File: part1_section2/types_demo.py
# Python: demonstrating a dynamic, strong type system
x = 10          # x is bound to an integer
print("x is", x, "of type", type(x).__name__)

x = "hello"     # the same name is rebound to a string; allowed at runtime
print("x is now", x, "of type", type(x).__name__)

# Python is strongly typed: it will NOT silently add a number to a string.
try:
    result = "The total is " + 5
    print(result)
except TypeError as e:
    print("TypeError caught:", e)

File: part1_section2/types_demo.js
// JavaScript: demonstrating a dynamic, weakly typed system
let x = 10;              // number
console.log("x is", x, "of type", typeof x);

x = "hello";            // reassigned to a string; allowed
console.log("x is now", x, "of type", typeof x);

// JavaScript is weakly typed: it silently coerces types.
let result = "The total is " + 5;   // number becomes a string
console.log(result, "->", typeof result);

console.log("5" - 2, "-> subtraction coerces the string to a number");
console.log("5" + 2, "-> addition coerces the number to a string");
